### PR TITLE
Get all PostTags and Add tag to Post 

### PR DESCRIPTION
**Pull request by angelabarber**
_**GETTING POST-TAGS**_

You will need to have PostTags added to your database--
```SQL
INSERT INTO PostTags (post_id, tag_id) VALUES (2, 1);
INSERT INTO PostTags (post_id, tag_id) VALUES (3,3);
INSERT INTO PostTags (post_id, tag_id) VALUES (3,1);
```

1. Navigate to your `server.py` and run the program.
2. In Postman- GET http://localhost:8000/post-tags
3. After pressing send you should get a `200` (the information will vary from my return, but this is an example of how it should look)
```
    {
        "post_tag_id": 1,
        "post_id": 3,
        "tag_id": 3,
        "tag": {
            "id": 3,
            "label": "Python"
        }
    }, 
```

_**ADD A POST TAG**_

1. 1. Navigate to your `server.py` and run the program.
2. In Postman- POST http://localhost:8000/post-tags
3. In the Body- Raw- JSON enter:
```
 {
    "post_id": 3,
    "tag_id": 3

}
```
5. This should return a `201 Created` with {} in the body
6. If you are missing input information, for example the `tag_id` is left blank, it will return a `400 bad request` with a body that says "Insufficient Input Information"

### CODE REVIEW

File: views/post_tags.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                JOIN Tags t
+                    ON t.id = pt.tag_id
+             """
+        )
+        query_results = db_cursor.fetchall()
+
+        post_tags = []
+        for row in query_results:
+            post_tag = {
+                "post_tag_id": row["id"],
```
Comment (HewsonSchulz) [inline views/post_tags.py]: May want to change this to be just `"id"`.

File: server.py [hunk 2]
```diff
@@ -191,6 +196,28 @@ def do_POST(self):
...
+
+                    if successfully_added:
+                        return self.response(
+                            "{}", status.HTTP_201_SUCCESS_CREATED.value
+                        )
+                    else:
+                        return self.response(
+                            "Failed to create", status.HTTP_500_SERVER_ERROR.value
+                        )
+                except JSONDecodeError:
```
Comment (HewsonSchulz) [inline server.py]: Changing this to `except (JSONDecodeError, KeyError):` will solve the problem I mentioned.

**Review by HewsonSchulz — CHANGES_REQUESTED** [review]
Everything works perfectly except for when "you are missing input information, for example the `tag_id`", it does not return `400 Bad Request` as stated, and instead throws a `KeyError`. The error doesn't occur when leaving the request body blank, but it does occur when attempting to use an empty object (`{}`).

File: views/post_tags.py (hunk 1 above)
Comment (angelabarber) [reply views/post_tags.py]: I have pushed the updated version that (I think) fixes those error issues and adjusts the PostTags id to just say id

**Review by HewsonSchulz — APPROVED** [review]
Everything looks good, and functions as intended.

**Pull request merged**